cluster/clustered_service_agent.go

```diff
@@ -663,12 +665,28 @@ func (agent *ClusteredServiceAgent) awaitRecordingId(sessionId int32) (int64, er
 			}
 			return false
 		})
+		if _, err := arch.PollForErrorResponse(); err != nil {
```

Comment:
in Java this is after checkForClockTick, should we move it there for consistency sake?

Reply:
https://github.com/real-logic/aeron/blob/154532e71d57fb90b51111c3caae02da4d10f47e/aeron-cluster/src/main/java/io/aeron/cluster/service/ClusteredServiceAgent.java#L1043-L1055

There is 1 check for arch error here in the counter as well. I guess they did 2 checks (1 in the above, and 1 below checkForClockTick - which I missed). But since the golang version combined counter and record into 1. I think we'll do 1 single check as you suggested will suffice.

Reply:
Understood